{
    "name": "@nivinjoseph/n-exception",
    "version": "1.0.16",
    "description": "Library of Exception classes",
    "main": "./dist/index.js",
    "types": "./dist/index.d.ts",
    "scripts": {
        "ts-compile": "tsc -p .",
        "ts-lint": "tslint -p . -c tslint.json",
        "ts-build": "npm run ts-compile && npm run ts-lint",
        "ts-build-dist": "npm run ts-build && tsc -p ./dist",
        "pretest": "npm run ts-build",
        "test": "mocha --reporter spec --ui tdd --require ts-node/register './test/**/*.test.js' || true",
        "pretest-ci": "npm run pretest",
        "test-ci": "mocha --reporter spec --ui tdd --require ts-node/register './test/**/*.test.js'",
        "publish-package": "npm run ts-build-dist && git add . && git commit -m 'preparing to publish new version' && npm version patch && git push && npm publish --access=public"
    },
    "repository": {
        "type": "git",
        "url": "git+https://github.com/nivinjoseph/n-exception.git"
    },
    "keywords": [
        "exception",
        "error"
    ],
    "author": "NiviN",
    "license": "MIT",
    "bugs": {
        "url": "https://github.com/nivinjoseph/n-exception/issues"
    },
    "homepage": "https://github.com/nivinjoseph/n-exception#readme",
    "devDependencies": {
        "@types/mocha": "^5.2.5",
        "@types/node": "^8.9.5",
        "mocha": "^5.2.0",
        "ts-node": "^7.0.1",
        "tslint": "^5.11.0",
        "typescript": "^3.1.1"
    },
    "dependencies": {
        "@nivinjoseph/n-ext": "^1.1.2"
    }
}
